File: android/app/src/main/java/vn/team/freechat/adapter/MessageListAdapters.java
package vn.team.freechat.adapter;

import android.content.Context;

import java.util.HashMap;
import java.util.Map;

/**
 * Created by tavandung12 on 10/6/18.
 */

public final class MessageListAdapters {

    private final Map<Long, MessageListAdapter> apdapters;
    private final static MessageListAdapters INSTANCE = new MessageListAdapters();

    private MessageListAdapters() {
        this.apdapters = new HashMap<>();
    }

    public static MessageListAdapters getInstance() {
        return INSTANCE;
    }

    public MessageListAdapter getAdapter(Context context, long targetChannelId) {
        synchronized (apdapters) {
            MessageListAdapter adapter = apdapters.get(targetChannelId);
            if(adapter == null) {
                adapter = new MessageListAdapter(context);
                apdapters.put(targetChannelId, adapter);
            }
            return adapter;
        }
    }

}

File: kotlin/app/src/main/java/vn/team/freechat_kotlin/adapter/MessageListAdapters.kt
package vn.team.freechat_kotlin.adapter


import android.content.Context
import java.util.*

/**
 * Created by tavandung12 on 10/6/18.
 */

class MessageListAdapters {

    private val apdapters: MutableMap<String, MessageListAdapter>

    private constructor() {
        this.apdapters = HashMap()
    }

    companion object {
        private val INSTANCE = MessageListAdapters()
        fun getInstance() : MessageListAdapters = INSTANCE
    }

    fun getAdapter(context: Context, targetName: String): MessageListAdapter {
        synchronized(apdapters) {
            var adapter: MessageListAdapter? = apdapters[targetName]
            if (adapter == null) {
                adapter = MessageListAdapter(context)
                apdapters.put(targetName, adapter)
            }
            return adapter
        }
    }

}